stdlib/public/core/StringLegacy.swift

```diff
@@ -33,14 +33,27 @@ extension String {
       self = count == 0 ? "" : repeatedValue
       return
     }
+    guard !repeatedValue.isEmpty else {
+        self = ""
+        return
+    }
 
-    // TODO(String performance): We can directly call appendInPlace
-    var result = String()
-    result.reserveCapacity(repeatedValue._guts.count &* count)
-    for _ in 0..<count {
-      result += repeatedValue
+    var repeatedValue = repeatedValue
+    let repeatedValueUTF8Count = repeatedValue.withUTF8 { $0.count }
+    self.init(_uninitializedCapacity: repeatedValueUTF8Count * count) {
+      buffer in
+      var total = 0
+      repeatedValue.withUTF8 { repeatedValueUTF8 in
```

Comment:
This is the second call to `repeatedValue.withUTF8`: could we consolidate into one call only?

Reply:
`self.init` can't be dispatched from inside of `.withUTF8`, so everything can't be put inside of the first `.withUTF8`. 
The first `.withUTF8` is needed to get count.
According to documentation second call of `.withUTF8` will not need allocations and should be fast.

Reply:
Is the first call different from `repeatedValue.utf8.count`?

Reply:
You could do `self = repeatedValue.withUTF8 { ...` and return the string from there.

Reply:
endeed!